halo2_proofs/src/plonk/mv_lookup.rs

```diff
@@ -0,0 +1,97 @@
+use super::circuit::Expression;
+use ff::Field;
+use std::fmt::{self, Debug};
+
+pub(crate) mod prover;
+pub(crate) mod verifier;
+
+/// Degree of lookup without inputs
+pub fn base_degree(table_degree: usize) -> usize {
+    // let lhs_degree = table_degree + inputs_expressions_degree + 1
+    // let degree = lhs_degree + 1
+    std::cmp::max(3, table_degree + 2)
+}
+
+pub fn degree_with_input(base_degree: usize, input_expression_degree: usize) -> usize {
+    base_degree + input_expression_degree
+}
+
+#[derive(Clone)]
+pub struct Argument<F: Field> {
+    pub(crate) table_expressions: Vec<Expression<F>>,
+    pub(crate) inputs_expressions: Vec<Vec<Expression<F>>>,
```

Comment:
[critical-2]
we use `let input = compress(lookup.input_expressions());`  in [snark-verifier](https://github.com/scroll-tech/snark-verifier/blob/4466059ce9a6dfaf26455e4ffb61d72af775cf52/snark-verifier/src/system/halo2.rs#L637) that expected `Vec<Expression<F>>` (not `Vec<Vec<Expression<F>>>`). During the integration test, i fake it by using  `let input = compress(lookup.table_expressions());`, so we can generate the proof, but can not pass the verification.

Reply:
I think now the corresponding snark-verifier code for mvlookup hasn't been implemented.

Reply:
Changes to `snark-verifier` is implemented in https://github.com/scroll-tech/snark-verifier/pull/24.